admin_guide/upgrading_citus.rst

```diff
@@ -7,67 +7,242 @@ $$$$$$$$$$$$$$$
+
+  You should see a "Clusters are compatible" message. If you do not, fix any errors before proceeding. Please ensure that
+
+  * :code:`NEW_PG_DATA` contains an empty database initialized by new PostgreSQL version
+  * The Citus extension **IS NOT** created
+
+5. Stop the old server.
+
+6. Perform the upgrade (like before but without the :code:`--check` option).
+
+  .. code-block:: bash
+
+    $NEW_PG_PATH/bin/pg_upgrade -b $OLD_PG_PATH/bin/ -B $NEW_PG_PATH/bin/ \
+                                -d $OLD_PG_DATA -D $NEW_PG_DATA
+
+7. Start the new server.
+
+8. Restore metadata.
+
+  .. code-block:: postgres
+
+    INSERT INTO pg_catalog.pg_dist_partition SELECT * FROM public.pg_dist_partition;
+    INSERT INTO pg_catalog.pg_dist_shard SELECT * FROM public.pg_dist_shard;
+    INSERT INTO pg_catalog.pg_dist_shard_placement SELECT * FROM public.pg_dist_shard_placement;
+    INSERT INTO pg_catalog.pg_dist_node SELECT * FROM public.pg_dist_node;
+    TRUNCATE TABLE pg_catalog.pg_dist_local_group;
+    INSERT INTO pg_catalog.pg_dist_local_group SELECT * FROM public.pg_dist_local_group;
+    INSERT INTO pg_catalog.pg_dist_transaction SELECT * FROM public.pg_dist_transaction;
+    INSERT INTO pg_catalog.pg_dist_colocation SELECT * FROM public.pg_dist_colocation;
+
+9. Drop temporary metadata tables.
+
+  .. code-block:: postgres
+
+    DROP TABLE public.pg_dist_partition;
+    DROP TABLE public.pg_dist_shard;
+    DROP TABLE public.pg_dist_shard_placement;
+    DROP TABLE public.pg_dist_node;
+    DROP TABLE public.pg_dist_local_group;
+    DROP TABLE public.pg_dist_transaction;
+    DROP TABLE public.pg_dist_colocation;
+
+10. Restart sequences.
+
+  .. code-block:: postgres
+
+    SELECT setval('pg_catalog.pg_dist_shardid_seq', (SELECT MAX(shardid)+1 AS max_shard_id FROM pg_dist_shard), false);
+
+    SELECT setval('pg_catalog.pg_dist_groupid_seq', (SELECT MAX(groupid)+1 AS max_group_id FROM pg_dist_node), false);
+
+    SELECT setval('pg_catalog.pg_dist_node_nodeid_seq', (SELECT MAX(nodeid)+1 AS max_node_id FROM pg_dist_node), false);
+
+    SELECT setval('pg_catalog.pg_dist_shard_placement_placementid_seq', (SELECT MAX(placementid)+1 AS max_placement_id FROM pg_dist_shard_placement), false);
+
+    SELECT setval('pg_catalog.pg_dist_colocationid_seq', (SELECT MAX(colocationid)+1 AS max_colocation_id FROM pg_dist_colocation), false);
+
+11. Register triggers.
+
+  .. code-block:: postgres
+
+    CREATE OR REPLACE FUNCTION create_truncate_trigger(table_name regclass) RETURNS void LANGUAGE plpgsql as $$
+    DECLARE
+      command  text;
+      trigger_name text;
+
+    BEGIN
+      trigger_name := 'truncate_trigger_' || table_name::oid;
+      command := 'create trigger ' || trigger_name || ' after truncate on ' || table_name || ' execute procedure pg_catalog.citus_truncate_trigger()';
+      execute command;
+      command := 'update pg_trigger set tgisinternal = true where tgname
+     = ' || trigger_name;
```

Comment:
I had to change this to ```quote_literal(trigger_name)```, else it gave me this error:
```
postgres=# SELECT create_truncate_trigger(logicalrelid) FROM pg_dist_partition ;
ERROR:  column "truncate_trigger_16623" does not exist
LINE 2:  = truncate_trigger_16623
           ^
QUERY:  update pg_trigger set tgisinternal = true where tgname
 = truncate_trigger_16623
CONTEXT:  PL/pgSQL function create_truncate_trigger(regclass) line 12 at EXECUTE
```

@begriffs Did you get similar error when you tried this out?